android-components/components/browser/menu2/src/main/java/mozilla/components/browser/menu2/BrowserMenuController.kt

```diff
@@ -55,11 +52,20 @@ class BrowserMenuController(
             style?.let { setStyle(it) }
         }
 
+        // Monitor for changes to the parent layout and dismiss pop up if displayed, else the menu
+        // could be displayed in the wrong position. For example, if the menu is displayed and the
+        // device orientation changes from portrait to landscape or vice versa.
```

Comment:
Note: This is also consistent with the 'settings' menu popup

Reply:
@rocketsroger 
I found a solid [reasoning](https://github.com/mozilla-mobile/fenix/pull/17682#issuecomment-769883653) why the hosts (fragments, controllers) handle the dismissal of the menus (like [here](https://searchfox.org/mozilla-mobile/source/fenix/app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt#1475)).  

I was wondering why the same approach hasn't been used towards bookmarks fragment or tabs tray? They keep the `BrowserMenu` at the same position, and Bookmarks fragment actually chops a good chunk from the menu list. Could it be a universal approach that we want to hide menus after the configuration change because of misalignment?

Reply:
As discussed with @mavduevskiy, I think in this case we're not using `OnConfigurationChanged` but using layout to determine if we should dismiss the menu.  However, we could add a parameter in `show()` to allow the user to determine that it should "auto dismiss"?